### PR TITLE
Approximating Simulation into master

### PR DESCRIPTION
**Pull request by hillmich**

### CODE REVIEW

File: include/ShorSimulator.hpp [hunk 1]
```diff
@@ -77,6 +77,9 @@ class ShorSimulator : public Simulator {
     const bool emulate;
     const bool verbose;
     const bool approximate;
+    unsigned long long approximation_runs{0};
+    long double final_fidelity{1.0L};
+    double step_fidelity{0.9};
```
Comment (burgholzer) [inline include/ShorSimulator.hpp:82]: Is it on purpose that these to have different types? I'd suppose `double` is enough for both of them.

File: scripts/approximation.sh [hunk 2]
```diff
@@ -85,6 +85,7 @@ sem -j 2 benchmark simfid69 --ps --step_fidelity 1.0 --simulate_shor 69 --simula
 sem -j 2 benchmark simfid55 --ps --step_fidelity 1.0 --simulate_shor 55 --simulate_shor_coprime 2
 sem -j 2 benchmark simfid33 --ps --step_fidelity 1.0 --simulate_shor 33 --simulate_shor_coprime 5
 
+
```
Comment (burgholzer) [inline scripts/approximation.sh:88]: unnecessary whitespace change
```suggestion
```

File: src/ShorSimulator.cpp [hunk 3]
```diff
@@ -119,7 +117,7 @@ std::map<std::string, unsigned int> ShorSimulator::Simulate(unsigned int shots)
         }
 
         if (approximate && (i + 1) % mod == 0) {
-            fidelity *= ApproximateByFidelity(0.9, false, true);
+            final_fidelity *= ApproximateByFidelity(step_fidelity, false, true);
```
Comment (burgholzer) [inline src/ShorSimulator.cpp:120]: for some reason this line is not covered by any tests. could you add a unit test for this?

**Review by burgholzer — COMMENTED** [review]
Some minor comments. Should be quick to fix.

File: include/ShorSimulator.hpp (hunk 1 above)
Comment (hillmich) [reply include/ShorSimulator.hpp:82]: `double` should be enough and I'll change that for the next bigger approximation change, but for now I'd like to keep this for consistency.

File: scripts/approximation.sh (hunk 2 above)
Comment (hillmich) [reply scripts/approximation.sh:88]: :+1:

File: src/ShorSimulator.cpp (hunk 3 above)
Comment (hillmich) [reply src/ShorSimulator.cpp:120]: Good catch, I'll a test.

File: include/ShorSimulator.hpp (hunk 1 above)
Comment (burgholzer) [reply include/ShorSimulator.hpp:82]: Sounds good 👍🏼
